x/upgrade/keeper/keeper_test.go

```diff
@@ -48,6 +58,182 @@ func (s *KeeperTestSuite) TestReadUpgradeInfoFromDisk() {
 	s.Require().Equal(expected, ui)
 }
 
+func (s *KeeperTestSuite) TestScheduleUpgrade() {
+	clientState := &ibctmtypes.ClientState{ChainId: "gaiachain"}
+	cs, err := clienttypes.PackClientState(clientState)
+	s.Require().NoError(err)
+
+	altClientState := &ibctmtypes.ClientState{ChainId: "ethermint"}
+	altCs, err := clienttypes.PackClientState(altClientState)
+	s.Require().NoError(err)
+
+	cases := []struct {
+		name    string
+		plan    types.Plan
+		setup   func()
+		expPass bool
+	}{
+		{
+			name: "successful time schedule",
+			plan: types.Plan{
+				Name: "all-good",
+				Info: "some text here",
+				Time: s.ctx.BlockTime().Add(time.Hour),
+			},
+			setup:   func() {},
+			expPass: true,
+		},
+		{
+			name: "successful height schedule",
+			plan: types.Plan{
+				Name:   "all-good",
+				Info:   "some text here",
+				Height: 123450000,
+			},
+			setup:   func() {},
+			expPass: true,
+		},
+		{
+			name: "successful ibc schedule",
+			plan: types.Plan{
+				Name:                "all-good",
+				Info:                "some text here",
+				Height:              123450000,
+				UpgradedClientState: cs,
+			},
+			setup:   func() {},
+			expPass: true,
+		},
+		{
+			name: "successful overwrite",
+			plan: types.Plan{
+				Name:   "all-good",
+				Info:   "some text here",
+				Height: 123450000,
+			},
+			setup: func() {
+				s.app.UpgradeKeeper.ScheduleUpgrade(s.ctx, types.Plan{
+					Name:   "alt-good",
+					Info:   "new text here",
+					Height: 543210000,
+				})
+			},
+			expPass: true,
+		},
+		{
+			name: "successful IBC overwrite",
+			plan: types.Plan{
+				Name:                "all-good",
+				Info:                "some text here",
+				Height:              123450000,
+				UpgradedClientState: cs,
+			},
+			setup: func() {
+				s.app.UpgradeKeeper.ScheduleUpgrade(s.ctx, types.Plan{
+					Name:                "alt-good",
+					Info:                "new text here",
+					Height:              543210000,
+					UpgradedClientState: altCs,
+				})
+			},
+			expPass: true,
+		},
+		{
+			name: "successful IBC overwrite with non IBC plan",
+			plan: types.Plan{
+				Name:   "all-good",
+				Info:   "some text here",
+				Height: 123450000,
+			},
+			setup: func() {
+				s.app.UpgradeKeeper.ScheduleUpgrade(s.ctx, types.Plan{
+					Name:                "alt-good",
+					Info:                "new text here",
+					Height:              543210000,
+					UpgradedClientState: altCs,
```

Comment:
this should overwrite with non IBC plan so the upgrade client state should be nil right?

Reply:
Correct, as mentioned in test case name

Reply:
I'm confused? Why is `UpgradedClientState: altCs` set then?